tests/v2/validation/networking/connectivity/port_test.go

```diff
@@ -0,0 +1,266 @@
+//go:build (validation || infra.rke2k3s || cluster.any || sanity) && !stress && !extended
+
+package connectivity
+
+import (
+	"fmt"
+	"math/rand/v2"
+	"net/url"
+	"strconv"
+	"strings"
+	"testing"
+
+	projectsapi "github.com/rancher/rancher/tests/v2/actions/projects"
+	"github.com/rancher/rancher/tests/v2/actions/services"
+	"github.com/rancher/rancher/tests/v2/actions/workloads"
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	steveV1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/rancher/shepherd/extensions/charts"
+	"github.com/rancher/shepherd/extensions/clusters"
+	kubeapinodes "github.com/rancher/shepherd/extensions/kubeapi/nodes"
+	shepworkloads "github.com/rancher/shepherd/extensions/workloads"
+	namegen "github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/rancher/shepherd/pkg/session"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	clusterNameAnnotation = "cluster.x-k8s.io/cluster-name"
+	portLimit             = 65534
+)
+
+type PortTestSuite struct {
+	suite.Suite
+	client      *rancher.Client
+	session     *session.Session
+	cluster     *management.Cluster
+	clusterName string
+}
+
+func (p *PortTestSuite) TearDownSuite() {
+	p.session.Cleanup()
+}
+
+func (p *PortTestSuite) SetupSuite() {
+	p.session = session.NewSession()
+
+	client, err := rancher.NewClient("", p.session)
+	require.NoError(p.T(), err)
+
+	p.client = client
+
+	log.Info("Getting cluster name from the config file and append cluster details in connection")
+	p.clusterName = client.RancherConfig.ClusterName
+	require.NotEmptyf(p.T(), p.clusterName, "Cluster name to install should be set")
+
+	clusterID, err := clusters.GetClusterIDByName(p.client, p.clusterName)
+	require.NoError(p.T(), err, "Error getting cluster ID")
+
+	p.cluster, err = p.client.Management.Cluster.ByID(clusterID)
+	require.NoError(p.T(), err)
+}
+
+func (p *PortTestSuite) TestHostPort() {
+	subSession := p.session.NewSession()
+	defer subSession.Cleanup()
+
+	log.Info("Creating new project and namespace")
+	_, namespace, err := projectsapi.CreateProjectAndNamespace(p.client, p.cluster.ID)
+	require.NoError(p.T(), err)
+
+	steveClient, err := p.client.Steve.ProxyDownstream(p.cluster.ID)
+	require.NoError(p.T(), err)
+
+	//This must be a valid port number, 0 < hostPort < 65536
+	//Using a random port to avoid 'port in use' failures and allow the test to be rerun
+	hostPort := rand.IntN(65534) + 1
```

Comment:
maybe bump this out above the rancher-reserved ports just in case? i.e. avoid these https://ranchermanager.docs.rancher.com/getting-started/installation-and-upgrade/installation-requirements/port-requirements

Reply:
There’s a very small chance it will return one of these ports, but I can use a range of ports that aren’t reserved by Rancher